### ISSUE TITLE
Simple usage question

### ISSUE THREAD

**Issue by jennybc** [ropensci/gistr#47]
What's the easiest path to get this Gist into my hands using `gistr`?

https://gist.github.com/expersso/4ac33b9c00751fddc7f8#file-google_maps_with_rvest

I tried this

```
> browse("https://gist.github.com/expersso/4ac33b9c00751fddc7f8#file-google_maps_with_rvest")
Error: Not Found
```

I'd like to fork it or get that code on my clipboard or get in a local file. It's just an example of wanting to get someone else's Gist and play with it.

Needless to say, I think this is just a documentation gap. And I obviously know how to complete the task other ways, just thought I'd try to do it with `gistr` for kicks.

**Comment by jennybc** [comment]
There's no example for `browse()`. So I just guessed at what sort of input I could provide. And apparently I guessed wrong.

**Comment by sckott** [comment]
hi @jennybc - good question

right, `browse` just takes a gist class object

**Comment by sckott** [comment]
browse to the gist in your default browser 

```r
gist("4ac33b9c00751fddc7f8") %>% browse()
```

fork a gist and browse to it

```r
gist('4ac33b9c00751fddc7f8') %>% fork() %>% browse()
```

**Comment by jennybc** [comment]
What if I wanted to fork, pull to my machine, and open/run/edit?

**Comment by sckott** [comment]
What is the exact behavior you want?  

- pull down into memory, then execute one or more files (assuming they are executable)
- pull down gist, save files to disk, then run things from disk 
- pull down gist, open file in rstudio to edit

**Comment by jennybc** [comment]
I was expecting a way to pull down the gist (the original or my fork) into local file(s). And have the path returned/printed? So options 2 or 3 above.

**Comment by sckott** [comment]
okay

**Comment by sckott** [comment]
okay, on reinstall see `gist_save()`

**Comment by jennybc** [comment]
Cool! About to be sucked in to meetings but will try some time later today and close.

**Comment by sckott** [comment]
:+1:

**Comment by jennybc** [comment]
Excellent! It all worked. Here are a few suggestions, but feel free to ignore. I'll close anyway.

Consider providing a gist creation/registration function that accepts a browser URL. Digging the gist id out by hand is frustrating. I wrestled a lot with same thing over in googlesheets. The API wants one sort of info, but human beings have easy access to another sort. It's easy to translate between them though, so why not do it?

It's sad that the gist id is lost in the default file name from `gist_save()`. Seems like it would be good to build that into filename itself, if someone wanted to remember which Gist was the original source.

I didn't try it, but what happens if gist contains multiple files?

**Comment by sckott** [comment]
* can definitely accept a URL in addition to gist id
* file names: hmmm, but if you do push up a gist with those same files, then would you strip the gist ids out?  if not, if you create a new gist, the gist id would be different, and would be diff. from those ids in the file name, kinda confusing
* multiple files: it's vectorized, so if there's multiple files, it saves all, and if you choose to open them, opens all in e.g., rstudio

**Comment by jennybc** [comment]
Yeah, I'm not entirely sure what to do about the file naming. One can think about various round trip scenarios. But in my example, where I've copied someone else's Gist, I'd probably want to record the source id. So if you did anything about this, perhaps it should be controlled via an argument.

**Comment by sckott** [comment]
Okay, an argument makes sense I think. What about for each gist, create folder of the gist id, then put files in that folder?

**Comment by jennybc** [comment]
That's a nice idea. You can still have the filenames as they are in the gist this way but, by virtue of the enclosing folder, retain the gist id? The folder to hold multiple files from same gist would be nice anyway.

**Comment by sckott** [comment]
@jennybc yeah, like 

```
- 4ac33b9c00751fddc7f8 -- # folder
                       | - file_name.R # file
                       | - file_name.md # file
```

**Comment by sckott** [comment]
@jennybc Okay, addressed the URL thing in #48 - reinstall and try passing in a gist URL  - has to be full URL - may want to support partial like 

* `gist.github.com/expersso/4ac33b9c00751fddc7f8`
* `expersso/4ac33b9c00751fddc7f8`
